### PR TITLE
Add package subcommand

### LINKED ISSUE #47: Add package subcommand
Currently the only way to produce a package is with the `publish` subcommand. This relies on access to a functioning Artifactory instance to work.

In some cases it may be useful to just produce the tgz and distribute it some other way.

### PR DESCRIPTION
**Pull request by asmello**
Fixes #47

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -41,6 +41,18 @@ enum Command {
...
+    /// Packages this api
+    #[clap(alias = "pack")]
+    Package {
+        /// Destination directory for the release
+        #[clap(long)]
+        #[arg(default_value = ".")]
+        directory: String,
+        /// Generate package but do not write it to filesystem
+        #[clap(long)]
+        dry_run: bool,
```
Comment (mara-schulke) [inline src/main.rs:53]: I would say this is not needed – people can just delete the tgz

File: src/main.rs [hunk 2]
```diff
@@ -41,6 +41,18 @@ enum Command {
         package: PackageId,
     },
 
+    /// Packages this api
+    #[clap(alias = "pack")]
+    Package {
+        /// Destination directory for the release
+        #[clap(long)]
+        #[arg(default_value = ".")]
+        directory: String,
```
Comment (mara-schulke) [inline src/main.rs:50]: Can we name it `--output-directory`? Directory could refer to the package directory as well / could be misleading

File: src/main.rs [hunk 3]
```diff
@@ -41,6 +41,18 @@ enum Command {
         package: PackageId,
     },
 
+    /// Packages this api
```
Comment (mara-schulke) [inline src/main.rs:44]:
```suggestion
    /// Exports the current package into a distributable tgz archive
```

File: src/main.rs [hunk 4]
```diff
@@ -234,6 +247,23 @@ mod cmd {
         manifest.write().await
     }
 
+    pub async fn package(directory: String, dry_run: bool) -> eyre::Result<()> {
```
Comment (mara-schulke) [inline src/main.rs:250]: This one is missing a doc comment

**Review by mara-schulke — APPROVED** [review]

File: src/main.rs (hunk 2 above)
Comment (asmello) [reply src/main.rs:50]: Makes sense, will update

File: src/main.rs (hunk 1 above)
Comment (asmello) [reply src/main.rs:53]: I think this is nice to have if you just want to validate the manifest and don't want to risk overwriting something. Unless you feel it's not worth the complexity I'd prefer to keep it.

**Pull request merged**